components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/PulsarProducer.java

```diff
@@ -16,57 +16,72 @@
  */
 package org.apache.camel.component.pulsar;
 
-import java.util.Map;
-
 import org.apache.camel.Exchange;
 import org.apache.camel.Message;
 import org.apache.camel.NoTypeConversionAvailableException;
 import org.apache.camel.TypeConversionException;
-import org.apache.camel.component.pulsar.utils.message.PulsarMessageHeaders;
 import org.apache.camel.component.pulsar.utils.message.PulsarMessageUtils;
-import org.apache.camel.support.DefaultProducer;
+import org.apache.camel.impl.DefaultProducer;
 import org.apache.pulsar.client.api.Producer;
 import org.apache.pulsar.client.api.ProducerBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class PulsarProducer extends DefaultProducer {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(PulsarProducer.class);
+
     private final PulsarEndpoint pulsarEndpoint;
+    private Producer<byte[]> producer;
 
     public PulsarProducer(PulsarEndpoint pulsarEndpoint) {
         super(pulsarEndpoint);
-
         this.pulsarEndpoint = pulsarEndpoint;
     }
 
+    public static PulsarProducer create(final PulsarEndpoint pulsarEndpoint) {
```

Comment:
This is never used as far as I see.

Also the producer is already created in the PulsarEndpoint.

https://github.com/apache/camel/blob/master/components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/PulsarEndpoint.java#L60

Reply:
Sorry an issue with cherry-picking a commit has brought in some deleted code will re-remove